os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -275,65 +275,74 @@ static void up_dumpstate(void)
 
 	ustackbase = (uint32_t)rtcb->adj_stack_ptr;
 	ustacksize = (uint32_t)rtcb->adj_stack_size;
+
 	lldbg("sp:     %08x\n", sp);
 
-#ifdef CONFIG_ARCH_NESTED_IRQ_STACK_SIZE
-	/* Get the limits on the nested irq stack */
-	nestirqstkbase = (uint32_t)&g_nestedirqstkbase;
-	nestirqstksize = (CONFIG_ARCH_NESTED_IRQ_STACK_SIZE & ~3);
+	/* Show the interrupt number, handler and stack information
+	 * if we are in an interrupt handler
+	 */
 
-	lldbg("Nested IRQ stack:\n");
-	lldbg("  base: %08x\n", nestirqstkbase);
-	lldbg("  size: %08x\n", nestirqstksize);
+	if (current_regs) {
```

Comment:
This is not the right way to check if we were handling irq when the assert happened. The current_regs will always be non NULL at this point because any assert condition will be handled in the exception handler only. In order to check if we were in an irq before the assert occurred, we need to check if there is an irq context stored on the stack before the context of the current assert handler. If it is stored, then we have to extract the irq number from that stacked context.

Reply:
Modified the code and verified with memfault.